File: cpp/src/player_state.cpp
/****************************************************************************
 *
 * @author Dilshod Mukhtarov <dilshodm(at)gmail.com>
 * Oct 2024
 *
 ****************************************************************************/

#include "player_state.h"

namespace scorbit {
namespace detail {

PlayerState::PlayerState(sb_player_t player, sb_score_t score)
    : m_player(player)
    , m_score {score}
{
}

sb_player_t PlayerState::player() const
{
    return m_player;
}

sb_score_t PlayerState::score() const
{
    return m_score;
}

void PlayerState::setScore(sb_score_t score)
{
    m_score = score;
}

} // namespace detail
} // namespace scorbit

File: python/src/player_state.py
# python/src/player_state.py

class PlayerState:
    def __init__(self, player: int, score: int = 0):
        self.player: int = player
        self.score: int = score

    def __eq__(self, other):
        if not isinstance(other, PlayerState):
            return NotImplemented
        return self.player == other.player and self.score == other.score

    def __ne__(self, other):
        return not (self == other)